pkg/chunk/aws/metrics_autoscaling.go

```diff
@@ -57,7 +57,7 @@ func (cfg *MetricsAutoScalingConfig) RegisterFlags(f *flag.FlagSet) {
 	f.Int64Var(&cfg.TargetQueueLen, "metrics.target-queue-length", 100000, "Queue length above which we will scale up capacity")
 	f.Float64Var(&cfg.ScaleUpFactor, "metrics.scale-up-factor", 1.3, "Scale up capacity by this multiple")
 	f.StringVar(&cfg.QueueLengthQuery, "metrics.queue-length-query", defaultQueueLenQuery, "query to fetch ingester queue length")
-	f.StringVar(&cfg.ErrorRateQuery, "metrics.error-rate-query", defaultErrorRateQuery, "query to fetch error rates per table")
+	f.StringVar(&cfg.ThrottleQuery, "metrics.write-throttle-query", defaultThrottleRateQuery, "query to fetch throttle rates per table")
```

Comment:
It's a good & necessary change, but it will also break any CI/CD systems. @rsteneteg FYI when we're wondering what changed after this is merged.

Reply:
What about keeping the old `metrics.error-rate-query` parameter but add `DEPRECATED` in the help string, and have both set `cfg.ThrottleQuery`, and then add a check during after parsing the flags, to look if any deprecated param has been set, and warn about it?

Reply:
if the code that looks for usage of deprecated flags also had a metric for how many deprecated flags are used, we could have an alert for it that lets us know that we need to look at the logs and see which param has been deprecated so we get time to update our CI/CD pipeline before the change actually break stuff

Reply:
Reporting via a metric seems a fine change to `pkg/util/flagext/deprecated.go` but beyond the scope of this PR.